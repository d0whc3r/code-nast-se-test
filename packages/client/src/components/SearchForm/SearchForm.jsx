import './SearchForm.styles.css'

function SearchForm({ onSearch }) {
  const handleSubmit = (e) => {
    e.preventDefault()
    onSearch?.(Object.fromEntries(new FormData(e.target)))
  }

  return (
    <form className="flex flex-col pt-6 bg-white rounded" onSubmit={handleSubmit}>
      <div className="md:grid md:grid-cols-2">
        <div className="input-container md:mr-4">
          <label className="label" htmlFor="topic">Topic</label>
          <input className="input" type="text" id="topic" name="topic" placeholder="Topic to search" />
        </div>
        <div className="md:grid md:grid-cols-2">
          <div className="input-container">
            <label className="label" htmlFor="from">Start date</label>
            <input className="input" type="date" id="from" name="from" />
          </div>
          <div className="input-container">
            <label className="label" htmlFor="to">End date</label>
            <input className="input" type="date" id="to" name="to" />
          </div>
        </div>
      </div>
      <button className="button self-start" type="submit">Search</button>
    </form>
  )
}

export default SearchForm
