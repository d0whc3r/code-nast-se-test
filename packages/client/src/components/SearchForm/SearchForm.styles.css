.input-container {
    @apply mb-4 md:mr-2;
}

.label {
    @apply block mb-2 text-sm font-bold text-gray-700;
}

.input {
    @apply w-full px-3 py-2 text-sm leading-tight text-gray-700 border rounded shadow appearance-none focus:outline-none;
}
