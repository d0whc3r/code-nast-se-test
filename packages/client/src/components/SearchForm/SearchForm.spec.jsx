import { describe, expect, it, vi } from 'vitest'
import { act, fireEvent, render, screen } from '../../../test/utils'
import SearchForm from './SearchForm'

describe('SearchForm', () => {
  it('should render', () => {
    const { baseElement } = render(<SearchForm />)
    expect(screen.getByText('Search')).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })
  it('should search', async () => {
    const searchSpy = vi.fn()
    render(<SearchForm onSearch={searchSpy} />)
    await act(() => {
      fireEvent.click(screen.getByText('Search'))
    })
    expect(searchSpy).toHaveBeenCalledTimes(1)
    expect(searchSpy).toHaveBeenCalledWith({ topic: '', from: '', to: '' })
  })
})
