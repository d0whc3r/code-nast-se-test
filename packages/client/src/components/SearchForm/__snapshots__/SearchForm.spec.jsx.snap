// Vitest Snapshot v1

exports[`SearchForm > should render 1`] = `
<body>
  <div>
    <form
      class="flex flex-col pt-6 bg-white rounded"
    >
      <div
        class="md:grid md:grid-cols-2"
      >
        <div
          class="input-container md:mr-4"
        >
          <label
            class="label"
            for="topic"
          >
            Topic
          </label>
          <input
            class="input"
            id="topic"
            name="topic"
            placeholder="Topic to search"
            type="text"
          />
        </div>
        <div
          class="md:grid md:grid-cols-2"
        >
          <div
            class="input-container"
          >
            <label
              class="label"
              for="from"
            >
              Start date
            </label>
            <input
              class="input"
              id="from"
              name="from"
              type="date"
            />
          </div>
          <div
            class="input-container"
          >
            <label
              class="label"
              for="to"
            >
              End date
            </label>
            <input
              class="input"
              id="to"
              name="to"
              type="date"
            />
          </div>
        </div>
      </div>
      <button
        class="button self-start"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</body>
`;
